src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -338,20 +286,29 @@ def setup(self, builder: Builder):
         self.step_size = builder.time.step_size()  # in days
         self.randomness = builder.randomness.get_stream(self.name)
 
+    def register_observations(self, builder: Builder) -> None:
+        # While this is a concatenating observation, it is not registered as such
+        # because we need to modify the per-time-step results_gatherer method
+        builder.results.register_unstratified_observation(
+            name=self.output_name,
+            pop_filter='alive == "alive" and tracked == True',
+            requires_columns=self.columns_required,
+            results_gatherer=self.get_observation,
+            results_updater=ConcatenatingObservation.concatenate_results,
+            to_observe=self.to_observe,
+        )
+
     def to_observe(self, event: Event) -> bool:
         """Only observe if Jan 1 occurs during the time step"""
         survey_date = dt.datetime(event.time.year, 1, 1)
         return self.clock() <= survey_date < event.time
 
-    def get_observation(self, event: Event) -> pd.DataFrame:
-        pop = self.population_view.get(
-            event.index,
-            query="alive == 'alive'",  # WIC should include only living simulants
-        )
+    def get_observation(self, pop: pd.DataFrame) -> pd.DataFrame:
         pop["income"] = self.pipelines["income"](pop.index)
 
         # add columns for output
-        pop["year"] = event.time.year
+        pop["simulant_id"] = pop.index
+        pop["year"] = pop["event_time"].dt.year
```

Comment:
There is some sort of helper function you could put here and all the other places you need to get random pop columns to reduce some of the duplicated code. Maybe its more trouble than its worth but there is somethign there.